pages/slice/language-reference/slice-grammar.md

```diff
@@ -2,14 +2,576 @@
 title: Slice grammar
 ---
 
+This page describes the grammatical rules, usage, and restrictions of the Slice language.
+
+## Slice elements
+
+### Slice files
+
+The `SliceFile` grammar rule is the parser's entry point.
+Once a file's contents have been [preprocessed], the resulting text is matched against this rule, in its entirety.
+
+While the contents of a Slice file may reference definitions in other files, each file is parsed separately.
+There is no notion of 'including' the contents of one file in another, and errors in one file do not affect the parsing of other files.
+
+A slice file consists of any number of [file attributes][attribute] and up to one [mode statement][mode-statement].
+These can appear in any order, but specifying more than one mode in a single file is forbidden.
+This is followed by an optional [module declaration][module-declaration], followed by any number of Slice definitions.
+
+```ebnf {% showTitle=false %}
+SliceFile
+    : SliceFilePrelude Module? Definition*
+    ;
+
+SliceFilePrelude
+    : EMPTY
+    | SliceFilePrelude FileCompilationMode
+    | SliceFilePrelude FileAttribute
+    ;
+
+Definition
+    : Struct | Class | Exception | Interface | Enum | CustomType | TypeAlias
+    ;
+```
+
+For additional information on Slice files, see the [Slice file][slice-file-guide] page.
+
+### Mode declarations
+
+Mode declarations consist of the `mode` keyword, followed by an equals sign, followed by an identifier.
+That identifier must be a valid Slice compilation mode - either `Slice1` or `Slice2`.
+
+```ebnf {% showTitle=false %}
+FileCompilationMode
+    : "mode" "=" Identifier
+    ;
+```
+
+Mode declarations must come before any Slice definitions or [module declarations][module-declaration].
+While the compilation mode you select will affect which Slice features are available to you,
+it has no effect on the parsing of, or validity of syntax within, your Slice file.
+For example, even though classes are a Slice1-only feature, `class` is still a keyword within a Slice2 file,
+and if you define a class, syntax errors in that class's definition will still be reported normally.
```

Comment:
```suggestion
For example, even though classes are a Slice1-only feature, `class` is still a keyword within a Slice2 file. As such, if you define a class in a Slice2 file, syntax errors in that class's definition are still reported.
```
 
Maybe something like this .

Reply:
I'm open to re-wording this section, but I don't think "as such" makes sense for it.
It implies that `class being a keyword in Slice2` => `syntax errors are still reported for it`.
They're 2 separate examples of the same concept, one doesn't really follow from the other.